packages/checkbox-group/src/LionCheckboxIndeterminate.js

```diff
@@ -0,0 +1,110 @@
+import { html, css } from '@lion/core';
+import { LionCheckbox } from './LionCheckbox.js';
+
+export class LionCheckboxIndeterminate extends LionCheckbox {
+  static get properties() {
+    return {
+      /**
+       * Indeterminate state of the checkbox
+       */
+      indeterminate: {
+        type: Boolean,
+        reflect: true,
+      },
+    };
+  }
+
+  get _checkboxGroupNode() {
+    return /** @type {import('./LionCheckboxGroup').LionCheckboxGroup} */ (this.parentElement);
+  }
+
+  get _subCheckboxes() {
+    return this._checkboxGroupNode.formElements.filter(checkbox => checkbox !== this);
+  }
+
+  _parentModelValueChanged() {
+    const checkedElements = this._subCheckboxes.filter(checkbox => checkbox.checked);
+    switch (this._subCheckboxes.length - checkedElements.length) {
+      // all checked
+      case 0:
+        this.indeterminate = false;
+        this.checked = true;
+        break;
+      // none checked
+      case this._subCheckboxes.length:
+        this.indeterminate = false;
+        this.checked = false;
+        break;
+      default:
+        this.indeterminate = true;
+    }
+  }
+
+  _ownInputChanged() {
+    this._subCheckboxes.forEach(checkbox => {
+      // eslint-disable-next-line no-param-reassign
+      checkbox.checked = this._inputNode.checked;
+    });
+  }
+
+  /**
+   * @override
+   * clicking on indeterminate status will set the status as checked
+   */
+  // eslint-disable-next-line class-methods-use-this
+  __toggleChecked() {}
```

Comment:
The current behavior is that it sometimes checks all checkboxes and sometimes unchecks all.

Maybe we can try to implement either one of those?
- a toggle between mixed, true and false like here: https://www.w3.org/TR/wai-aria-practices-1.1/examples/checkbox/checkbox-2/checkbox-2.html
- readonly, like here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox

Reply:
Added a small improvement to my branch to reuse the __toggleChecked method. I believe the behaviour is already correct though. If not, please clarify what behaviour is the current and what you expected to be different.

Reply:
Don't know if it's linked, but when I programmatically change the checked attribute of a children checkbox, it's not called (which is the expected behavior). So, I need to also register on this event. What's the best way to do it ?

Reply:
That might be a bug then. Does it work when you change the checked attribute of a child checkbox native input type="checkbox" node?

Reply:
Yea ... that's a tricky one. We only respond to `change` event on the native checkbox input node through proxy at the moment. If we want to also catch programmative changes we should listen to `model-value-changed` event which bubbles up, and we can then check `ev.detail.formPath` to see where it originated from, to determine what to do.

I will do a small refactor today to see if I can make that work 👍

Reply:
I pushed a commit for this, it now uses model-value-changed. The logic to determine what to do depending on where the event comes from and to ensure we don't undo our own checked state setting was a bit tricky but it ended up acceptable to me ;)